move/interchain_token_service/sources/types/coin_info.move

```diff
@@ -21,14 +25,27 @@ module interchain_token_service::coin_info {
         }
     }
 
-    /// Create a new coin info from the given `CoinMetadata` object.
+    /// Create a new coin info from the given `CoinMetadata` object and publicly freeze the metadata object.
     public fun from_metadata<T>(metadata: CoinMetadata<T>): CoinInfo<T> {
-        CoinInfo {
+        let coin_info = CoinInfo {
             name: metadata.get_name(),
             symbol: metadata.get_symbol(),
             decimals: metadata.get_decimals(),
-            metadata: option::some(metadata),
-        }
+            metadata: option::none(),
+        };
+        transfer::public_freeze_object(metadata);
+        coin_info
+    }
+
+    // -----------------
+    // Package Functions
+    // -----------------
+
+    /// Publicly freeze metadata for a coin from the given `CoinInfo<T>`
+    public(package) fun release_metadata<T>(coin_info: &mut CoinInfo<T>) {
+        assert!(coin_info.metadata.is_some(), EMetadataEmpty);
```

Comment:
for simplicity of operations, make this a no-op if the metadata was already none. Don't want one token_id to cause the whole batch to fail, and we don't need to filter by token_id type beforehand this way